Core/clim-basic/clipboard.lisp

```diff
@@ -0,0 +1,76 @@
+(in-package :climi)
+
+;;;
+;;;  Functions dealing with copying to the clipboard.
+;;;
+
+(deftype representation-type-name () '(member :string :html :image))
+
+(defgeneric copy-to-clipboard-with-port (port sheet clipboard-p object presentation-type)
+  (:documentation "Method to be implemented by backends."))
```

Comment:
documentation should say, what this function is supposed to do, not who should implement it. you may put that in a comment (and update documentation).

Most notably I do not understand what is clipboard-p, why it needs both port an sheet etc.

Reply:
`CLIPBOARD-P` is a boolean specifying whether the function should operate on the clipboard or the selection. Perhaps having a keyword argument `:CLIPBOARD` or `:SELECTION` would be better? Another alternative is to split it into two separate functions.

The reason both `PORT` and `SHEET` is specified is because the generic function needs to dispatch on the port. Note that this function should never be called by users.

The user-facing API consists of two functions: `COPY-TO-CLIPBOARD` and `COPY-TO-SELECTION`. Both of these functions then simply call `COPY-TO-CLIPBOARD-WITH-PORT`, with the first argument simply being the sheet's port, and `CLIPBOARD-P` being used to discriminate which of the previously mentioned two functions were called.

I agree that all of this information should go into the docstring. I'If you're OK with the API, then I'll write the documentation. Otherwise I'll change the API first.

Reply:
please see my other comment.